httpipfs.go

```diff
@@ -119,6 +130,23 @@ func (hi *HttpIpfs) ServeHTTP(res http.ResponseWriter, req *http.Request) {
 
 	selNode := unixfsnode.UnixFSPathSelectorBuilder(path.String(), dagScope.TerminalSelectorSpec(), false)
 
+	sig := RequestSignature{
+		requestId: req.Header.Get("X-Signature"),
+		cid:       rootCid.String(),
+		protocol:  "https",
+	}
+	b, err := json.Marshal(sig)
+	if err != nil {
+		logError(http.StatusInternalServerError, err)
+		return
+	}
+	sigSigned, err := hi.privKey.Sign(b)
+	if err != nil {
+		logError(http.StatusInternalServerError, err)
+		return
+	}
+	res.Header().Set("X-Signature", string(sigSigned))
```

Comment:
Please note we need to include both the payload we are signing and the signature. Otherwise it would be tricky to verify the signature - the verifier would have to reconstruct the JSON-encoded payload with exact formatting as used by Frisbii.

In https://www.notion.so/pl-strflt/Content-retrieval-attestation-a6df8198d5a248ceac4c004ee971759c?pvs=4#069d2dd99bf94bb08e33ea7636e1f332, I am proposing the following:

> Next, the storage provider signs this payload using its private key. Finally, the payload and the signature are encoded using base64 and concatenated into a single attestation string `PAYLOAD.SIGNATURE`

Also:

> Discussion point: should we follow the [JWT](https://jwt.io/introduction) and UCAN format that have three segments - HEADER.PAYLOAD.SIGNATURE? The header includes metadata like the version of the format used.

Here is the description of a JWT header:


> The header typically consists of two parts: the type of the token, which is JWT, and the signing algorithm being used, such as HMAC SHA256 or RSA.
> 
> For example:
>
> ```json
> {
>  "alg": "HS256",
>  "typ": "JWT"
> }
>```
>
> Then, this JSON is Base64Url encoded to form the first part of the JWT.

Reply:
gotcha, I changed it to `PAYLOAD.SIGNATURE`. `PAYLOAD` is the json encoded string, and can include dots too. Do we need to escape that, or is only the last dot (before `SIGNATURE`) significant?

Reply:
Shall we base64 encode `PAYLOAD`?

Reply:
f791a89